types/underscore/index.d.ts

```diff
@@ -3721,14 +3721,14 @@ declare module _ {
          * @returns The item on the `object` or the `defaultValue`
          **/
         get<V extends Collection<any>>(
-            object: V,
+            object: V | null | undefined,
             path: string | string[]
         ): TypeOfCollection<V> | undefined;
         get<V extends Collection<any>, U>(
-            object: V,
+            object: V | null | undefined,
             path: string | string[],
             defaultValue?: U
-        ): TypeOfCollection<V> | U;
+        ): TypeOfCollection<V> | U | undefined;
```

Comment:
If you want to be precise, you can use separate overloads for more specific return types:

```ts
        get(object: null | undefined, path: string | string[]): undefined;
        get<V extends Collection<any>>(
            object: V,
            path: string | string[]
        ): TypeOfCollection<V> | undefined;
        get<U>(
            object: null | undefined,
            path: string | string[],
            defaultValue?: U
        ): U;
        get<V extends Collection<any>, U>(
            object: V,
            path: string | string[],
            defaultValue?: U
        ): TypeOfCollection<V> | U;
```